src/routes/safe/store/middleware/notificationsMiddleware.js

```diff
@@ -24,9 +27,15 @@ const notificationsMiddleware = (store: Store<GlobalState>) => (next: Function)
         const transactionsList = action.payload
         const userAddress: string = userAccountSelector(state)
         const awaitingTransactions = getAwaitingTransactions(transactionsList, userAddress)
-
+        const safes = safesMapSelector(state)
 
         awaitingTransactions.map((awaitingTransactionsList, safeAddress) => {
+          const safe = safes.get(safeAddress)
+          // If the user is not an owner, do not send notifications
+          if (!isUserOwner(safe, userAddress)) {
```

Comment:
This check shouldn't be inside .map, you could check it right after you get `safes` and `userAddress`.

Reply:
Cannot do that, I dont have the current `safeAddress`, I tried with the `safeParamAddressSelector` selector but It does not work inside a middleware. Do you have any alternative ideas?

Reply:
You can try to modify `safeParamAddressSelector` so it uses router state from redux or create a new selector

Reply:
Done